dist/doc/spec.adoc

```diff
@@ -0,0 +1,1270 @@
+
+* operands of the concatenation operator (`&`) shall be coerced to a string, except when an operand is an array. Arrays shall be coerced to an array of strings.
+* operands of numeric operators (`+`, `-`, `*`, `/`) shall be coerced to numbers except when the operand is an array. Arrays shall be coerced to an array of numbers.
+* operands of the union operator (`~`) shall be coerced to an array
+* The left-hand operand of ordering comparison operators (`>`, `>=`, `<`, `\<=`) must be a string or number.  Any other type shall be coerced to a number.
+* If the operands of an ordering comparison are different, they shall both be coerced to a number
+* parameters to functions shall be coerced to the expected type as defined by the function signature
+* slice and flatten operations shall coerce operands to a number
+
+The equality and inequality operators (`=`, `==`, `!=`, `<>`) do **not** perform type coercion.  If operands are different types, the values are considered not equal.
+
+Coercion is not always possible, and if so, an error shall be emitted -- most often an `invalid-type` error.
+
+==== Examples
+
+[source%unbreakable]
+----
+    search("abc" & 123, {}) -> "abc123"
+    search("123" * 2, {}) -> 246
+    search([1,2,3] ~ 4, {}) -> [1,2,3,4]
+    search(123 < "124", {}) -> true
+    search("23" > 111, {}) -> false
+    search(abs("-2"), {}) -> 2
+    search([1,2,3,4]["1"], {}) -> 2
+    search(1 == "1", {}) -> false
+----
+
+=== Type Coercion Rules
+
+[%header,cols="1,1,1"]
+|===
+| Provided Type | Expected Type | Result
+| number | string | number converted to a string. Similar to JavaScript `toString()`
+| boolean | string | "true" or "false"
+| array | string | Not supported
+| object | string | Not supported
+| null | string | "" (empty string)
+| string | number | Parse string to a number.  If the string is not a well-formed number, will return 0.  Allow for locale-specific currency symbols to be ignored.
+| boolean | number | true => 1 false => 0
+| array | number | Not supported
+| object | number | Not supported
+| null | number | 0
+| number | array | create a single-element array with the number
+| string | array | create a single-element array with the string.
+| boolean | array | create a single-element array with the boolean.
+| object | array | Not supported
+| null | array | Empty array
+| number | object | No supported
+| string | object | Not supported
+| boolean | object | Not supported
+| array | object | An object where the keys are the array index positions and the values are the array values.
+| null | object | Empty object
+|===
+
+==== Examples
+
+[source%unbreakable]
+----
+    search("\"$123.00\" + 1", {}) -> 124.00"
+    search("truth is " & `true`, {}) -> "truth is true"
+    search(2 + `true`, {}) -> 3
+    search(avg("20"), {}) -> 20
+    search(merge({a: 3}, 4), {}) -> {a:3}
+    search(merge({a: 3}, ["x", "y"]), {}) -> {"0": "x", "1": "y", "a": 3}
```

Comment:
sounds good.  I have added all the samples from the spec to a new test: specSamples.spec.js so that if any of the results change we will know to update the spec